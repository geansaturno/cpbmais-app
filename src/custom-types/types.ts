export interface Menu {
    label: string
    link?: string
    internal?: Menu[]
}
