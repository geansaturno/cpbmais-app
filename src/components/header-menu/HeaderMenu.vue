<template>
  <ul class="menu menu__wrapper ">
    <li
      v-for="item in menu"
      :key="item.label"
      class="menu__item"
    >
      <router-link :to="item.link" :is="item.link ? 'router-link': 'span'" class="">
        {{item.label}}
        <ul v-if="item.internal">
          <li
            v-for="internal in item.internal"
            :key="internal.name"
          >
            <router-link :to="internal.link" :is="internal.link ? 'router-link': 'span'" class="menu__internal">
              {{internal.label}}
            </router-link>
          </li>
        </ul>
      </router-link>
    </li>
  </ul>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'
import { MenuItem } from '@/custom-types'

@Component({
  name: 'HeaderMenu'
})
export default class HelloWorld extends Vue {
  @Prop({ required: true })
  menu!: MenuItem;
}
</script>

<style lang="scss">
@import 'src/assets/scss/variables.scss';

.menu {
  background-color: $g3;
  max-width: 350px;
  width: 100vw;

  &__item {
    color: $g1;
    font-size: 21px;
    padding: 19px 0;
    margin: 0 22px;

    &:not(:last-child) {
      border-bottom: 1px solid $g2;
    }
  }

  &__wrapper {
    list-style: none;
    padding-left: 0;

    &--internal {
      padding-left: 30px;
    }
  }
}

</style>
