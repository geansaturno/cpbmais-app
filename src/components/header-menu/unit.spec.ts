import 'jest'
import { mount } from '@vue/test-utils'
import HeaderMenu from './HeaderMenu.vue'
import { Menu } from '@/custom-types'

describe('HeaderMenu.vue', () => {
  it('deve exibir o menu', () => {
    const menu: Menu[] = [
      {
        label: 'Home',
        link: '/'
      },
      {
        label: 'biblioteca',
        link: '/lib'
      },
      {
        label: 'Meditações',
        internal: [
          {
            label: 'Até que ele volte',
            link: '/meditacoes/ate-que-ele-volte'
          },
          {
            label: 'Sublime beleza',
            link: '/meditacoes/sublime-beleza'
          }
        ]
      }
    ]

    const wrapper = mount(HeaderMenu, {
      propsData: {
        menu
      }
    })

    expect(wrapper).toMatchSnapshot()
  })
})
