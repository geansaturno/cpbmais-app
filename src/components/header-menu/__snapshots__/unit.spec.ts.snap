// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderMenu.vue deve exibir o menu 1`] = `
<ul>
  <li><a href="/">
      Home

      <!---->
    </a></li>
  <li><a href="/lib">
      biblioteca

      <!---->
    </a></li>
  <li><span>
      Meditações

      <ul><li><a href="/meditacoes/ate-que-ele-volte">
            Até que ele volte
          </a></li><li><a href="/meditacoes/sublime-beleza">
            Sublime beleza
          </a></li></ul></span></li>
</ul>
`;
